lua/rainbow-delimiters/stack.lua

```diff
@@ -98,12 +103,12 @@ end
 function M.new(items)
 	---@type Stack
 	local result = {
-		content = {},  ---@type any[]
-		size = size,   ---@type fun(self: Stack): number
-		iter = iter,
-		push = push,   ---@type fun(self: Stack, item: any): Stack
-		pop  = pop,    ---@type fun(self: Stack): any
-		peek = peek,   ---@type fun(self: Stack): any
+		content = {}, ---@type any[]
+		size    = size, ---@type fun(self: Stack): integer
+		iter    = iter, ---@type fun(self: Stack): ((fun(i: integer, item: any): integer?, any), Stack, integer)
+		push    = push, ---@type fun(self: Stack, item: any): Stack
+		pop     = pop, ---@type fun(self: Stack): any
+		peek    = peek, ---@type fun(self: Stack): any
```

Comment:
Do we still need these type annotations? I think we can remove them, the functions and the fields are all type annotated already.

```suggestion
		content = {},
		size    = size,
		iter    = iter,
		push    = push,
		pop     = pop,
		peek    = peek,
```

Reply:
I think you are right, we don't need these. I only added the last one since the others were already there.